### PR TITLE
WIP: Adding Logging 

### PR DESCRIPTION
**Pull request by mvanniasingheTT**
This WIP PR looks to add logging of raw stats. Currently tested with offline inference. I have the logger class in `vllm-tt-metal-llama3-70b/src/logging_utils.py` but I also have it under `tests` right now so you can see how it is used. Any feedback would be great thank you!

### CODE REVIEW

**Comment by tstescoTT** [comment]
If there is no decode in the generation, the logging will error out:
```
Traceback (most recent call last):                                                                                                                                                                                                               │····
  File "benchmark_vllm_offline_inference.py", line 207, in <module>                                                                                                                                                                              │····
    run_inference(                                                                                                                                                                                                                               │····
  File "benchmark_vllm_offline_inference.py", line 116, in run_inference                                                                                                                                                                         │····
    run_inference_perf(llm, prompt_token_ids, sampling_params)                                                                                                                                                                                   │····
  File "benchmark_vllm_offline_inference.py", line 139, in run_inference_perf                                                                                                                                                                    │····
    generate_tokens(                                                                                                                                                                                                                             │····
  File "benchmark_vllm_offline_inference.py", line 168, in generate_tokens                                                                                                                                                                       │····
    outputs = llm.generate(prompts, sampling_params, prompt_token_ids)                                                                                                                                                                           │····
  File "/home/user/vllm/vllm/utils.py", line 1073, in inner                                                                                                                                                                                      │····
    return fn(*args, **kwargs)                                                                                                                                                                                                                   │····
  File "/home/user/vllm/vllm/entrypoints/llm.py", line 353, in generate                                                                                                                                                                          │····
    outputs = self._run_engine(use_tqdm=use_tqdm)                                                                                                                                                                                                │····
  File "/home/user/vllm/vllm/entrypoints/llm.py", line 879, in _run_engine                                                                                                                                                                       │····
    step_outputs = self.llm_engine.step()                                                                                                                                                                                                        │····
  File "/home/user/vllm/vllm/engine/llm_engine.py", line 1454, in step                                                                                                                                                                           │····
    self.do_log_stats(scheduler_outputs, outputs)                                                                                                                                                                                                │····
  File "/home/user/vllm/vllm/engine/llm_engine.py", line 1562, in do_log_stats                                                                                                                                                                   │····
    logger.log(stats)                                                                                                                                                                                                                            │····
  File "/home/user/tests/mock_vllm_model.py", line 428, in log                                                                                                                                                                                   │····
    self._write_to_json(stats)                                                                                                                                                                                                                   │····
  File "/home/user/tests/mock_vllm_model.py", line 458, in _write_to_json                                                                                                                                                                        │····
    data["time to first token"][                                                                                                                                                                                                                 │····
KeyError: 'Inference num:1'  
```

Can you make the logging handle this case?

File: vllm-tt-metal-llama3-70b/src/logging_utils.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+            ]
+            self.time_per_output_token.append(tpot)  # Add all values to the list
+
+        self._write_to_json(stats)
+
+    def _write_to_json(self, stats):
+        data = {}
+
+        if len(stats.time_per_output_tokens_iter) > 0:
+            data["time per output token"] = {}
```
Comment (tstescoTT) [inline vllm-tt-metal-llama3-70b/src/logging_utils.py:118]: For dictionary keys could you use snake_case e.g. `data["time to first token"]` -> `data["time_to_first_token"]` or `data["ttft"]`?

File: vllm-tt-metal-llama3-70b/.env.llama31_8b [hunk 2]
```diff
@@ -0,0 +1,24 @@
+MODEL_NAME=llama-3.1-8b-instruct
```
Comment (tstescoTT) [inline vllm-tt-metal-llama3-70b/.env.llama31_8b:1]: Can you add this file to .gitignore?

**Comment by tstescoTT** [comment]
Could you remove add test/data and add the .jsonl output to .gitignore?

File: vllm-tt-metal-llama3-70b/src/logging_utils.py (hunk 1 above)
Comment (mvanniasingheTT) [reply vllm-tt-metal-llama3-70b/src/logging_utils.py]: yes, it is resolved now

**Pull request closed without merging**